bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -71,11 +80,15 @@
     final long ledgerId;
     long lastAddPushed;
     volatile long lastAddConfirmed;
+    volatile long explicitLastAddConfirmed;
+    volatile long piggyBackedLastAddConfirmed;
+
     long length;
     final DigestManager macManager;
     final DistributionSchedule distributionSchedule;
 
     final RateLimiter throttler;
+    private ScheduledExecutorService lastAddConfirmedUpdateScheduler = null;
```

Comment:
I think @sijie and @eolivelli are suggesting not to create ScheduledExecutorService for each ledger, instead use shared ExecutorServices preferably bk.mainWorkerPool (OrderedSafeExecutor)